pkgs/development/libraries/libdeltachat/default.nix

```
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, perl
+, pkg-config
+, rustPlatform
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libdeltachat";
+  version = "1.52.0";
+
+  src = fetchFromGitHub {
+    owner = "deltachat";
+    repo = "deltachat-core-rust";
+    rev = version;
+    sha256 = "0im2wmnxjw6j09aca68ma1fn23vr728gp41c9clp3jf08bcfnfzm";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "0qfja0qkq976vk3210qbrm3820d1v3zvzra2387dyphyswqjqjm6";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    perl
+    pkg-config
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  meta = with lib; {
+    description = "Delta Chat Rust Core library";
+    homepage = "https://github.com/deltachat/deltachat-core-rust/";
+    changelog = "https://github.com/deltachat/deltachat-core-rust/blob/${version}/CHANGELOG.md";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ dotlambda ];
```

```suggestion
    maintainers = with maintainers; [ dotlambda ];
    # darwin fails with:
    # error: failed to add native library /private/tmp/nix-build-libdeltachat-1.52.0.drv-0/source/target/release/build/libsqlite3-sys-e85b24d5407a705c/out/libsqlite3.a: file too small to be an archive
    broken = stdenv.isDarwin;
```

feel free to short the error message.

---

@SuperSandro2000 Can you try again?

---

I didn't build it myself. Ofborg did it for me.

---

I ended up setting `platforms = platforms.linux`. If someone wants to use it on macOS, feel free to contribute a patch.